examples/memstore-plugin/main.go

```diff
@@ -71,10 +72,14 @@ func main() {
 	opentracing.SetGlobalTracer(tracer)
 
 	memStorePlugin := grpcMemory.NewStoragePlugin(memory.NewStore(), memory.NewStore())
-	grpc.ServeWithGRPCServer(&shared.PluginServices{
+	service := &shared.PluginServices{
 		Store:        memStorePlugin,
 		ArchiveStore: memStorePlugin,
-	}, func(options []googleGRPC.ServerOption) *googleGRPC.Server {
+	}
+	if v.GetBool("enable_streaming_writer") {
```

Comment:
wouldn't it be simpler to declare this flag next to L41 instead of going through the configuration file?

Reply:
If using the same level of flag as `config`, it looks like we also need to modify `plugin/storage/grpc/config/config.go` (add code to parse this flag, then pass the flag in `buildPlugin`). This is similar to the previous `grpc-storage-plugin.writer-type` flag, but transfers the use of the flag to the plugin author.
If I understand correctly, we want to automatically upgrade the rpc method (don't want to explicitly add a flag to the user), but want to test both rpc methods in the integration test.
Now it seems that there are several options, which one do you prefer?
Option 1: Add a flag similar to `grpc-storage-plugin.writer-type` and let the plugin (plugin server) handle this flag.
Option 2: Add a flag like `grpc-storage-plugin.writer-type`, the default is auto. Handle this flag in jaeger (plugin client).
Option 3: Read the flag from config in the integration test, which is the current code.
Or do you have any other suggestions?

Reply:
Option 4 - add `flag.BoolPtr` in this file (`examples/memstore-plugin/main.go`) (instead of reading it through config file / viper) and use it in L79.

Reply:
If I understand correctly, then we also need to parse and transparently transmit this flag to the plugin in [`plugin/storage/grpc/config/config.go:119` ](https://github.com/jaegertracing/jaeger/blob/main/plugin/storage/grpc/config/config.go#L119), because **the flag passed in [`plugin/storage/integration/grpc_test.go:152`](https://github.com/jaegertracing/jaeger/blob/main/plugin/storage/integration/grpc_test.go#L152) is for jaeger, not for then plugin**. Do you think this is okay?

Reply:
Oh, I see what you mean. Ok let's leave it with the config file then.